### PR TITLE
prov/gni: implement fi_setopt and fi_getopt for scalable endpoints

### LINKED ISSUE #4994: prov/gni: fi_setopt and fi_getopt not implemented for scalable endpoints
Attempts to call fi_getopt() or fi_setopt() with a gni scalable endpoint fail with Function Not Implemented.   I talked to hppritcha, he believes this was just an oversight.

### PR DESCRIPTION
**Pull request by krehm**
Add support for fi_getopt and fi_setopt for scalable endpoints.
Needed so that FI_OPT_MIN_MULTI_RECV can be set appropriately for
multi-recv endpoints.

fixes #4994

Signed-off-by: Kevan Rehm <krehm@cray.com>

### CODE REVIEW

File: prov/gni/include/gnix_ep.h [hunk 1]
```diff
@@ -416,6 +416,42 @@ int gnix_pep_bind(fid_t fid, struct fid *bfid, uint64_t flags);
...
+ * @param[out] optval   the value of the named option
+ * @param[out] optlen   the width of the value in bytes
+ *
+ * @return FI_SUCCESS	upon successfully returning the value of the option
+ * @return -FI_EINVAL   missing or invalid fid, optval, and/or optlen
+ * @return -FI_ENOENT   unsupported context type
+ * @return -FI_ENOPROTOOPT  unknown option level or name
+ */
+
+int gnix_getopt(fid_t fid, int level, int optname,
```
Comment (hppritcha) [inline prov/gni/include/gnix_ep.h]: why did you change the function name from ```gnix_ep_getopt``` to ```gnix_getopt```?  we've usually used a qualifier after the ```gnix``` string to indicate what type of libfab object the function operates on.

**Comment by hppritcha** [comment]
@krehm except for the function name change, and the need to update the cray copyright at the top of the 3 files changed, this looks good.

File: prov/gni/include/gnix_ep.h (hunk 1 above)
Comment (krehm) [reply prov/gni/include/gnix_ep.h]: I followed the model used for gnix_cancel.  When getopt is called, there are two separate paths taken, one for regular 'ep' endpoints and one for scalable 'sep' endpoints.  The gnix_ep_getopt() name was retained for use with a regular endpoint, and the gnix_sep_getopt() name was used for a scalable endpoint.   Like gnix_cancel, I then used gnix_getopt() for the routine which decides which subroutine to call.  Does this make sense?

Comment (krehm) [reply prov/gni/include/gnix_ep.h]: Duh, there is no gnix_sep_getopt() routine, I'm not doing well today.   But I did follow the gnix_cancel model.

The alternative would be to change the name to gnix_ep_getopt, and then the routine that it calls would have to be something like __gnix_ep_getopt() I guess.   Is that what you'd like?

Comment (hppritcha) [reply prov/gni/include/gnix_ep.h]: hmm.. i see your point about the routine names.  so let's stick with what you have.  Could you update the copyright though?

Comment (krehm) [reply prov/gni/include/gnix_ep.h]: Done.

**Pull request merged**